### PR TITLE
net: lib: nrf_cloud: align poll timeout with mqtt keepalive

### PR DESCRIPTION
**Pull request by jhn-nordic**
Poll timeout of cloud socket is now aligned with the mqtt
keepalive.

Signed-off-by: Jon Helge Nistad <jon.helge.nistad@nordicsemi.no>

### CODE REVIEW

**Review by jtguggedal — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

File: subsys/net/lib/nrf_cloud/src/nrf_cloud.c [hunk 1]
```diff
@@ -535,7 +535,8 @@ void nrf_cloud_run(void)
 	atomic_set(&transport_disconnected, 0);
 
 	while (true) {
-		ret = poll(fds, ARRAY_SIZE(fds), POLL_TIMEOUT_MS);
+		ret = poll(fds, ARRAY_SIZE(fds),
+			cloud_keepalive_time_left(nrf_cloud_backend));
 
 		if (ret == 0) {
 			if (cloud_keepalive_time_left(nrf_cloud_backend) <
```
Comment (simensrostad) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud.c:542]: AFAIK poll returns 0 when the timeout expires. Then this check is redundant and the define ```POLL_TIMEOUT_MS``` can be removed.

**Review by simensrostad — APPROVED** [review]
Please include this change to the ```aws_iot.c``` cloud backend as well.

**Review by jayteemo — COMMENTED** [review]
The separate (shorter) poll timeout was added because if the socket is closed by another thread, the poll function will not return until the poll timeout value expires.   If the MQTT timeout value is long, then it would be a long wait before the disconnect notification.
Specifically, when the application calls `cloud_disconnect()` its gets to `mqtt_disconnect()` which calls `close()` on the socket.  If `mqtt_disconnect()` called `shutdown()` on the socket, I think poll would return immediately.

Is this change fixing a problem?

If you decide it is necessary, test asset tracker with a long MQTT keep alive value (20 minutes) and delete/re-add a device to your account to verify the desired behavior.

**Comment by jhn-nordic** [comment]
Thanks for your comment @jayteemo . The reason behind this change was that the short poll timeout wakes up the mcu and gives extra current draw. In applications that report data at >10 min interval, this added current is significant.

**Comment by simensrostad** [comment]
This PR is superseded by https://github.com/nrfconnect/sdk-nrf/pull/3472.

**Pull request closed without merging**